core/src/test/java/org/apache/iceberg/TestManifestCaching.java

```diff
@@ -178,22 +181,15 @@ public void testWeakFileIOReferenceCleanUp() {
       }
     }
 
-    System.gc();
+    GcFinalization.awaitFullGc();
```

Comment:
the comment in L188 says `// Insert one more FileIO to trigger cache eviction.` but aren't we triggering cache eviction already here?
It seems the original version of the test was expecting things to be evicted only when `IO_MANIFEST_CACHE_MAX_FILEIO_DEFAULT` is reached/exceeded

Reply:
I think that comment I made is inherently flawed since I use `System.gc()`.

The main purpose of the test is to verify that cache entry with garbage collected keys will be removed from the cache, and `GcFinalization.awaitFullGc()` + `manifestCache.cleanUp()` does evict those entries. In [572f0ca](https://github.com/apache/iceberg/pull/5862/commits/572f0ca0322f0974ad0b31ac4c184bce8b91ef84), I simplified testWeakFileIOReferenceCleanUp by not exceeding IO_MANIFEST_CACHE_MAX_FILEIO_DEFAULT.